halo2_proofs/src/plonk/evaluation.rs

```diff
@@ -975,14 +975,15 @@ impl<C: CurveAffine> Evaluator<C> {
     }
 
     fn compute_cluster_idx(degree: usize, max_cluster_idx: usize) -> usize {
-        let mut idx = if degree != 0 {
-            (31 - (degree as u32).leading_zeros()) as usize
-        } else {
-            0
-        };
-        if 1 << idx < degree {
-            idx = idx + 1;
+        let mut idx = 0;
+        if degree != 0 {
+            // same as unstable degree.ilog()
+            idx = (degree as f32).log2().floor() as usize;
```

Comment:
Good catch with the versioning.

I would prefer `idx = (31 - (degree as u32).leading_zeros()) as usize` over introducing floating-point math.

Reply:
NP, let's back to the origin code :)